### ISSUE TITLE
vila_live application is not running properly.

### ISSUE THREAD

**Issue by Shehjad-Ishan** [nvidia-holoscan/holohub#465]
For some reason the vila_live application not behaving as expected. I don't understand if it's a web related issue. The model checkpoints are being loaded fine but can't sync with the local server.  After several restarts I can get it to work sometimes.

```
`./run launch vila_live
Default language for vila_live selected: cpp
Run environment: export PYTHONPATH=${PYTHONPATH}:/../../../python/lib:/workspace/holohub/build/vila_live/python/lib:/workspace/holohub && export HOLOHUB_DATA_PATH=/media/sigmind/URSTP_HDD1411/holohub/data
Run workdir: cd /workspace/holohub/build/vila_live
Run command: bash -c '/workspace/holohub/applications/vila_live/run_vila_live.sh'
Run command args: 
[command] export PYTHONPATH=${PYTHONPATH}:/../../../python/lib:/workspace/holohub/build/vila_live/python/lib:/workspace/holohub && export HOLOHUB_DATA_PATH=/media/sigmind/URSTP_HDD1411/holohub/data
[command] cd /workspace/holohub/build/vila_live
[command] bash -c '/workspace/holohub/applications/vila_live/run_vila_live.sh'
[info] [fragment.cpp:569] Loading extensions from configs...
[info] [gxf_executor.cpp:249] [V4L2 to VLM app] Creating context
[info] [gxf_executor.cpp:1912] Activating Graph...
[info] [gxf_executor.cpp:1944] [V4L2 to VLM app] Running Graph...
[info] [gxf_executor.cpp:1946] [V4L2 to VLM app] Waiting for completion...
2024-08-10 06:00:43.698 INFO  gxf/std/greedy_scheduler.cpp@191: Scheduling 4 entities
[info] [context.cpp:50] _______________
[info] [context.cpp:50] Vulkan Version:
[info] [context.cpp:50]  - available:  1.3.204
[info] [context.cpp:50]  - requesting: 1.2.0
[info] [context.cpp:50] ______________________
[info] [context.cpp:50] Used Instance Layers :
[info] [context.cpp:50] 
[info] [context.cpp:50] Used Instance Extensions :
[info] [context.cpp:50] VK_EXT_debug_utils
[info] [context.cpp:50] VK_KHR_external_memory_capabilities
[info] [context.cpp:50] ____________________
[info] [context.cpp:50] Compatible Devices :
[info] [context.cpp:50] 0: NVIDIA RTX A2000 12GB
[info] [context.cpp:50] Physical devices found : 
[info] [context.cpp:50] 1
[info] [context.cpp:50] ________________________
[info] [context.cpp:50] Used Device Extensions :
[info] [context.cpp:50] VK_KHR_external_memory
[info] [context.cpp:50] VK_KHR_external_memory_fd
[info] [context.cpp:50] VK_KHR_external_semaphore
[info] [context.cpp:50] VK_KHR_external_semaphore_fd
[info] [context.cpp:50] VK_KHR_push_descriptor
[info] [context.cpp:50] VK_EXT_line_rasterization
[info] [context.cpp:50] 
[info] [vulkan_app.cpp:845] Using device 0: NVIDIA RTX A2000 12GB (UUID ef9478313962e02ed94bf73bd881b32f)
-- starting webserver @ 0.0.0.0:8050
 * Serving Flask app 'webserver'
 * Debug mode: on
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8050
 * Running on http://192.168.0.101:8050
Press CTRL+C to quit
2024-08-10 06:00:47 | INFO | controller | args: Namespace(host='0.0.0.0', port=10000, dispatch_method='shortest_queue')
2024-08-10 06:00:47 | INFO | controller | Init controller
2024-08-10 06:00:47 | ERROR | stderr | INFO:     Started server process [355]
2024-08-10 06:00:47 | ERROR | stderr | INFO:     Waiting for application startup.
2024-08-10 06:00:47 | ERROR | stderr | INFO:     Application startup complete.
2024-08-10 06:00:47 | ERROR | stderr | INFO:     Uvicorn running on http://0.0.0.0:10000 (Press CTRL+C to quit)
[info] [holoviz.cpp:1449] Input spec:
- type: color
  name: ""
  opacity: 1.000000
  priority: 0

2024-08-10 06:00:56 | INFO | model_worker | args: Namespace(host='0.0.0.0', port=40000, worker_address='http://localhost:40000', controller_address='http://localhost:10000', model_type='LLaMa', model_path='/workspace/holohub/Llama-3-VILA1.5-8b-AWQ', model_name=None, quant_path='/workspace/holohub/Llama-3-VILA1.5-8b-AWQ/llm/llama-3-vila1.5-8b-w4-g128-awq-v2.pt', precision='W4A16', device='cuda', multi_modal=False, limit_model_concurrency=5, stream_interval=1, no_register=False)
2024-08-10 06:00:56 | INFO | model_worker | Loading the model Llama-3-VILA1.5-8b-AWQ-4bit-AWQ on worker 2f7ea0 ...
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
2024-08-10 06:00:56 | ERROR | stderr | /workspace/holohub/build/vila_live/tinychat/models/vila_llama.py:31: UserWarning: model_dtype not found in config, defaulting to torch.float16.
2024-08-10 06:00:56 | ERROR | stderr |   warnings.warn("model_dtype not found in config, defaulting to torch.float16.")
real weight quantization...(init only):   0%|                                                                                                                | 0/32 [00:00<?, ?it/s]
real weight quantization...(init only):   9%|█████████▊                                                                                              | 3/32 [00:00<00:01, 19.82it/s]
real weight quantization...(init only):  19%|███████████████████▌                                                                                    | 6/32 [00:00<00:01, 23.64it/s]
real weight quantization...(init only):  28%|█████████████████████████████▎                                                                          | 9/32 [00:00<00:00, 24.60it/s]
real weight quantization...(init only):  38%|██████████████████████████████████████▋                                                                | 12/32 [00:00<00:00, 25.62it/s]
real weight quantization...(init only):  47%|████████████████████████████████████████████████▎                                                      | 15/32 [00:00<00:00, 24.92it/s]
real weight quantization...(init only):  56%|█████████████████████████████████████████████████████████▉                                             | 18/32 [00:00<00:00, 23.27it/s]
real weight quantization...(init only):  66%|███████████████████████████████████████████████████████████████████▌                                   | 21/32 [00:00<00:00, 22.46it/s]
real weight quantization...(init only):  75%|█████████████████████████████████████████████████████████████████████████████▎                         | 24/32 [00:01<00:00, 24.33it/s]
real weight quantization...(init only):  84%|██████████████████████████████████████████████████████████████████████████████████████▉                | 27/32 [00:01<00:00, 25.11it/s]
real weight quantization...(init only):  94%|████████████████████████████████████████████████████████████████████████████████████████████████▌      | 30/32 [00:01<00:00, 25.70it/s]
real weight quantization...(init only): 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████| 32/32 [00:01<00:00, 24.52it/s]
2024-08-10 06:01:04 | ERROR | stderr | 
  0%|                                                                                                                                                         | 0/1 [00:00<?, ?it/s]
Loading checkpoint:   0%|                                                                                                                                     | 0/1 [00:00<?, ?it/s]
Loading checkpoint: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:39<00:00, 39.41s/it]
Loading checkpoint: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:39<00:00, 39.41s/it]
2024-08-10 06:01:43 | ERROR | stderr | 
Exception in thread Thread-4 (annotate_image):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/workspace/holohub/applications/vila_live/vila_live.py", line 57, in annotate_image
    for response in self.vlm.generate_response(prompt, image_b64):
  File "/workspace/holohub/applications/vila_live/vlm.py", line 68, in generate_response
    raise ConnectionError("VILA server unavailable after 60 seconds")
ConnectionError: VILA server unavailable after 60 seconds
2024-08-10 06:01:50 | INFO | model_worker | Register to controller
2024-08-10 06:01:50 | INFO | controller | Register a new worker: http://localhost:40000
2024-08-10 06:01:50 | INFO | controller | Register done: http://localhost:40000, {'model_names': ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ'], 'speed': 1, 'queue_length': 0}
2024-08-10 06:01:50 | INFO | stdout | INFO:     127.0.0.1:50308 - "POST /register_worker HTTP/1.1" 200 OK
2024-08-10 06:01:50 | ERROR | stderr | INFO:     Started server process [356]
2024-08-10 06:01:50 | ERROR | stderr | INFO:     Waiting for application startup.
2024-08-10 06:01:50 | ERROR | stderr | INFO:     Application startup complete.
2024-08-10 06:01:50 | ERROR | stderr | INFO:     Uvicorn running on http://0.0.0.0:40000 (Press CTRL+C to quit)
2024-08-10 06:02:05 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:02:05 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:02:05 | INFO | stdout | INFO:     127.0.0.1:54816 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2024-08-10 06:02:20 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:02:20 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:02:20 | INFO | stdout | INFO:     127.0.0.1:54748 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2024-08-10 06:02:35 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:02:35 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:02:35 | INFO | stdout | INFO:     127.0.0.1:40550 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2024-08-10 06:02:50 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:02:50 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:02:50 | INFO | stdout | INFO:     127.0.0.1:56454 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2024-08-10 06:03:05 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:03:05 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:03:05 | INFO | stdout | INFO:     127.0.0.1:50318 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2024-08-10 06:03:20 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:03:20 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:03:20 | INFO | stdout | INFO:     127.0.0.1:49738 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2024-08-10 06:03:35 | INFO | model_worker | Send heart beat. Models: ['Llama-3-VILA1.5-8b-AWQ-4bit-AWQ']. Semaphore: None. global_counter: 0
2024-08-10 06:03:35 | INFO | controller | Receive heart beat. http://localhost:40000
2024-08-10 06:03:35 | INFO | stdout | INFO:     127.0.0.1:51842 - "POST /receive_heart_beat HTTP/1.1" 200 OK

`
```

**Comment by NigelNelson** [comment]
Hello! Could you please specify what hardware you're running this with? Also, as a first attempt, could you try increasing the timeout to 120 seconds in `vlm.py` [here](https://github.com/nvidia-holoscan/holohub/blob/e1453b36a652682865d6d9d807d565435ca4f16f/applications/vila_live/vlm.py#L40).

**Comment by Shehjad-Ishan** [comment]
> Hello! Could you please specify what hardware you're running this with? Also, as a first attempt, could you try increasing the timeout to 120 seconds in `vlm.py` [here](https://github.com/nvidia-holoscan/holohub/blob/e1453b36a652682865d6d9d807d565435ca4f16f/applications/vila_live/vlm.py#L40).

I am on a x86_dGPU machine. So far increasing the timeout to 120s is working!

**Comment by NigelNelson** [comment]
Great to hear! I tested this app with an RTX A6000, which has a much wider memory bus (384-bit vs 128-bit). So, it likely just takes the A2000 longer to load the model into GPU memory.

**Comment by Shehjad-Ishan** [comment]
> Great to hear! I tested this app with an RTX A6000, which has a much wider memory bus (384-bit vs 128-bit). So, it likely just takes the A2000 longer to load the model into GPU memory.

Thanks for the solution. Also I let the app running for 2 days. I have a 32 GB memory which is 90 percent occupied as I am writing. Is the memory leaking from somewhere? As I have inspected the code seemed like the chat history is not being saved. I modified the code to show all the response generated by the model in the chat-history-container and using a loopback video as feed.
